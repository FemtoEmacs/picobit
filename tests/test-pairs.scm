(define l '(0 1 2 3 4 5 6))

(define l2 (cons 2 3))

(set-car! l2 1)
(display (car l2))

;; ;; (define l2 (cons (list-ref l 4) 2)) ; (list-ref l (getchar))

;(led l) ;; invalid, but just to use l

(define x 9)
;(led x)

(display (list-ref l 6)) ;; TODO but (car l2) is ok, cdr too, but l and l2 fail, l2 with a segfault, since it's a pair and not a list, list bugs too but not as spectacular
;; TODO louche, si on fait list-ref de 1, ca marche, 4 affiche rien, 6 segfaulte
