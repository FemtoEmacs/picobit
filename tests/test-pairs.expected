:0080000080
:10800000FBD72404811E400080040120811240001F
:10801000810E4000810C4000810D40008008011A53
:10802000804200028027011B8027011C8024011D43
:108030008071011F806E0121800A000280270122C9
:10804000807701238073012480240002806A00026B
:1080500080730125800601268067012780090111B0
:1080600080400113804001148032011580270116E1
:108070008066011780050118806901198078000267
:10808000807D010F806601108070010B8007010A5E
:108090008078010B02B401975202B401885002B4F7
:1080A0000182530607CD514105D041CEB58AEEA1DC
:1080B000050A4062B582EED80120DAB8954321DB8B
:1080C000B6800221CC9A21CE2161EE2022CFB67259
:1080D00000EF20B6800120DA9EA02607E620B558E2
:1080E000EEA02607E600EF20C0B8BF20B582B64854
:1080F000012004C8B8AEA03121B5A402B583CDDCFF
:10810000EF02A034210EC5C122CD2102CDB4011150
:108110007101210EC89220EF210E426221B6620148
:108120000421C2EF2002B559B75520CCB8B8A02C15
:1081300007E620CEB51FEE20CFB6800120D296A054
:108140002D07E600EF20CC9DA02407E620CEB50544
:10815000EE20CFB666A107B000B8EE20B000D5EE95
:10816000A02D07E600EF20D995A108B100B820F0B6
:108170009C2095A106B100B8A104B100B8A109B135
:1081800000B8012007E600EF022104C79320CEEFDC
:1081900020CF2205C24072022004C8212302CDCD87
:1081A000B401A471012204C822242302CDCDCDB490
:1081B00001B3710123B8DC21B0011F23222602CDB7
:1081C000CDCDB401C6710122B8C621B0011F212452
:1081D0002602CDCDCDB401D971012022242602CDB5
:1081E000CDCDCDB401E770002121C4232502CDCD32
:1081F000B401F4710122B893219220EF22980194E6
:1082000020B1011F20EF21B87BB77500B86EB76AA7
:0682100021B73B21B72558
:00000001FF
